### PR TITLE
`Text` utility class

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR adds the utility class `Text` for working with text lines and migrates `KnownTypes` to it.

An instance of `Text` represents immutable list of text lines separated by `System.lineSeparator()`. It is meant to replace usage of `ImmutableList<String>` in our code.

### CODE REVIEW

File: base/src/main/java/io/spine/util/Lines.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    public static String join(String[] lines) {
+        checkNotNull(lines);
+        return joiner().join(lines);
+    }
+
+    /**
+     * Obtains the {@link Joiner} on {@link System#lineSeparator()}.
+     */
+    public static Joiner joiner() {
+        return Joiner.on(lineSeparator());
```
Comment (armiol) [inline base/src/main/java/io/spine/util/Lines.java:78]: Please consider caching some bits. `joiner` is stateful, so it's not a candidate. But `lineSeparator()` looks like one.

**Review by armiol — APPROVED** [review]
LGTM. Please see my comment on performance. Given the potentially big number of invocations, it seems important.

**Comment by alexander-yevsyukov** [comment]
@armiol, having more look at what we do with test, I think we need `Text` utility class which does a bit more than what `Lines` do. I'll work more on this PR to introduce the utility container.

File: base/src/main/java/io/spine/util/Text.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        return Objects.equals(lines, strings.lines);
+    }
+
+    @Override
+    public int hashCode() {
+        return lines.hashCode();
+    }
+
+    /*
+     * Static utilities. Please prefer instance methods of {@code Text}.
```
Comment (armiol) [inline base/src/main/java/io/spine/util/Text.java:160]: Please format the section according to Spine guidelines. 

Or move this recommendation to some other place, which will be reflected in the rendered Javadocs.

**Review by armiol — APPROVED** [review]
@alexander-yevsyukov LGTM, but please see my comment prior to merging.

**Pull request merged**